### PR TITLE
Consolidate REE and RPMB filesystems

### PR DESCRIPTION
**Pull request by jforissier**
This supersedes https://github.com/OP-TEE/optee_os/pull/741. The patchset is exactly the same, but split into more manageable commits, and rebased on latest master.

### CODE REVIEW

File: core/tee/tee_ree_fs.c [hunk 1]
```diff
@@ -277,7 +268,7 @@ static int ree_fs_open(const char *file, int flags, ...)
 	return res;
 }
 
-static int ree_fs_read(int fd, void *buf, size_t len)
+static int ree_fs_read_ree(int fd, void *buf, size_t len)
```
Comment (pascal-brand-st-dev) [inline core/tee/tee_ree_fs.c]: Commit message wrongly indicates `rpmb_fs_*` instead of `ree_fs*`

**Comment by pascal-brand-st-dev** [comment]
With my very minor comment
`Reviewed-by: Pascal Brand <pascal.brand@linaro.org>`
`Tested-by: Pascal Brand <pascal.brand@linaro.org> (STM)`

File: core/tee/tee_ree_fs.c (hunk 1 above)
Comment (jforissier) [inline core/tee/tee_ree_fs.c]: Indeed, I'll fix it when I add the review tags.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jforissier** [comment]
Rebased with tags applied.

**Pull request merged**